plugin-desktopswitch/desktopswitchconfiguration.cpp

```diff
@@ -64,18 +65,25 @@ void DesktopSwitchConfiguration::loadSettings()
 
 void DesktopSwitchConfiguration::loadDesktopsNames()
 {
-    int n = KX11Extras::numberOfDesktops();
+    LXQtPanelApplication *a = reinterpret_cast<LXQtPanelApplication*>(qApp);
+    auto wmBackend = a->getWMBackend();
+
+    int n = wmBackend->getWorkspacesCount();
     for (int i = 1; i <= n; i++)
     {
-        QLineEdit *edit = new QLineEdit(KX11Extras::desktopName(i), this);
+        QLineEdit *edit = new QLineEdit(wmBackend->getWorkspaceName(i), this);
         ((QFormLayout *) ui->namesGroupBox->layout())->addRow(tr("Desktop %1:").arg(i), edit);
 
+        //TODO: on Wayland we cannot set desktop names in a standart way
+        // On KWin we could use DBus org.kde.KWin as done by kcm_kwin_virtualdesktops
+        edit->setReadOnly(true);
+
         // C++11 rocks!
-        QTimer *timer = new QTimer(this);
-        timer->setInterval(400);
-        timer->setSingleShot(true);
-        connect(timer, &QTimer::timeout,       this, [=] { KX11Extras::setDesktopName(i, edit->text()); });
-        connect(edit,  &QLineEdit::textEdited, this, [=] { timer->start(); });
+        //QTimer *timer = new QTimer(this);
+        //timer->setInterval(400);
+        //timer->setSingleShot(true);
+        //connect(timer, &QTimer::timeout,       this, [=] { KX11Extras::setDesktopName(i, edit->text()); });
+        //connect(edit,  &QLineEdit::textEdited, this, [=] { timer->start(); });
```

Comment:
Would you please explain why you removed this under X11 instead of limiting it to X11? Or is it added to another place which I can't find?

Reply:
Hi, this was a quick fix because there will be no Wayland equivalent of setting desktop name.
At first I though about removing it from X11 too to keep both at same level but it's good to keep it.